src/Take.Blip.Builder/NoOwnerScheduleDecorator.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Lime.Protocol;
+using Take.Blip.Client.Extensions.Scheduler;
+using Takenet.Iris.Messaging.Resources;
+
+namespace Take.Blip.Builder
+{
+    /// <summary>
+    /// Defines a decorator of <see cref="ISchedulerExtension"/> that remove <see cref="OwnerContext"/> for handling commands before sending.
+    /// </summary>
+    public class NoOwnerScheduleDecorator : ISchedulerExtension
+    {
+        private readonly ISchedulerExtension _schedulerExtension;
+
+        /// <summary>
+        /// OwnerContext to dispose and recreate
+        /// </summary>
+        public IDisposable OwnerDisposable { get; private set; }
+
+        /// <summary>
+        /// Identity of owner to recreate owner context
+        /// </summary>
+        public Identity OwnerIdentity { get; private set; }
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="schedulerExtension"></param>
+        /// <param name="ownerContext"></param>
+        /// <param name="ownerIdentity"></param>
+        public NoOwnerScheduleDecorator(ISchedulerExtension schedulerExtension, IDisposable ownerContext, Identity ownerIdentity)
+        {
+            _schedulerExtension = schedulerExtension;
+            OwnerDisposable = ownerContext;
+            OwnerIdentity = ownerIdentity;
+        }
+
+        /// <summary>
+        /// Cancel a message scheduled without owner
+        /// </summary>
+        /// <param name="messageId"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public async Task CancelScheduledMessageAsync(string messageId, CancellationToken cancellationToken = default)
+        {
+            OwnerDisposable?.Dispose();
+            await _schedulerExtension.CancelScheduledMessageAsync(messageId, cancellationToken);
+            OwnerDisposable = OwnerContext.Create(OwnerIdentity);
```

Comment:
here you are changing the reference `OwnerDisposable` and not the value which was created externally.

Reply:
As It was a singleton reference inside `OwnerDisposable`, I thought that this implementation would be enough

Reply:
Your suggestion pf change `ISchedulerExtension` is good, i will go in that way.